osu.Game/Screens/Play/HUD/JudgementCounter/JudgementTally.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Scoring;
+
+namespace osu.Game.Screens.Play.HUD.JudgementCounter
+{
+    public partial class JudgementTally : CompositeDrawable
```

Comment:
To confirm, is the whole point of this class is to create bindables for each of the judgement types?

I'm wondering why the counter can't just consume `ScoreProcessore` directly, ie. is there some other reason to add this layer of complexity?

Reply:
Nope, its purpose is more akin to the receptor for the keys in the keycounterdisplay class, or the "clickspersecondcalculator" , i cached it with the HUD overlay so that the values passed to the judgement counter display actually account for every judgement, not just those from the creation of that particular instance of the counter onwards, is there a better way of doing it?

Reply:
The alternative would be to access `ScoreInfo.Statistics` directly. But that doesn't provide bindables, so you'd be updating things in response to `NewJudgement` / `JudgementReverted` and it would likely be a bit messier.

I was mostly curious why you went down this path. I guess it was mostly just matching how the key counter was implemented?

Reply:
despite the keycounter making things click id actually attribute the knowledge that i had to be able to do this to my practice mode ventures, i learnt a lot about the player, player loader, HUD overlay etc.. logic and i just figured having it load itself in with the HUD as a sort of "hub" for info made sense i guess? 😅